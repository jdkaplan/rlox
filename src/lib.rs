macro_rules! debug_log_gc {
    ($($arg:tt)*) => {{
        #[cfg(feature = "log_gc")]
        println!($($arg)*);
    }};
}

mod chunk;
mod compiler;
mod gc;
mod heap;
mod object;
mod scanner;
mod table;
mod value;
mod vm;

pub use vm::{InterpretError, InterpretResult, Vm};

pub const U8_COUNT: usize = (u8::MAX as usize) + 1;
pub const FRAMES_MAX: usize = 64;
pub const STACK_MAX: usize = FRAMES_MAX * U8_COUNT;
