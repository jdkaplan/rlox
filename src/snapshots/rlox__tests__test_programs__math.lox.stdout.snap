---
source: src/lib.rs
description: "print 0.1 + 0.2;\n\nvar a1 = 1 + 2 * 3 - 4 / 5 * 6 + 7 - 8;\nvar a2 = (((1 + (2 * 3)) - ((4 / 5) * 6)) + 7) - 8;\nprint a1;\nprint a2;\nprint a1 == a2;\n\nvar b = (1 + 2) * (3 - 4 / 5) * 6 + 7 + (-8);\nprint b;\nprint b == 38.6;\n\nvar pos_inf =  1/0;\nvar neg_inf = -1/0;\nvar nan =      0/0;\n\nprint pos_inf;\nprint neg_inf;\nprint nan;\n\nprint \"NaN compares false with everything...\";\nprint nan < nan;\nprint nan > nan;\nprint nan <= nan;\nprint nan >= nan;\nprint nan < 0;\nprint nan > 0;\nprint nan <= 0;\nprint nan >= 0;\n\nprint \"... and is not-equal to anything\";\nprint nan == nan;\nprint nan == 0;\nprint nan != nan;\nprint nan != 0;\n"
input_file: test_programs/math.lox
---
0.30000000000000004
1.1999999999999993
1.1999999999999993
true
38.6
true
inf
-inf
NaN
NaN compares false with everything...
false
false
false
false
false
false
false
false
... and is not-equal to anything
false
false
true
true

